opm/parser/eclipse/Generator/KeywordGenerator.cpp

```diff
@@ -0,0 +1,212 @@
+
+        newSource << sourceHeader();
+        for (auto iter = loader.keyword_begin(); iter != loader.keyword_end(); ++iter) {
+            std::shared_ptr<ParserKeyword> keyword = (*iter).second;
+            newSource << keyword->createCode() << std::endl;
+        }
+        newSource << "}" << std::endl;
+        {
+            newSource << "void Parser::addDefaultKeywords() {" << std::endl;
+            for (auto iter = loader.keyword_begin(); iter != loader.keyword_end(); ++iter) {
+                std::shared_ptr<ParserKeyword> keyword = (*iter).second;
+                newSource << "   addKeyword<ParserKeywords::" << keyword->className() << ">();" << std::endl;
+            }
+            newSource << "}" << std::endl;
+        }
+        newSource << "}" << std::endl;
+
+        {
+            bool update = updateFile( newSource , sourceFile );
+            if (m_verbose) {
+                if (update)
+                    std::cout << "Updated source file written to: " << sourceFile << std::endl;
+                else
+                    std::cout << "No changes to source file: " << sourceFile << std::endl;
+            }
+            return update;
+        }
+    }
+
+
+    bool KeywordGenerator::updateHeader(const KeywordLoader& loader , const std::string& headerFile) const {
+        std::stringstream stream;
+
+        stream << headerHeader();
+        for (auto iter = loader.keyword_begin(); iter != loader.keyword_end(); ++iter) {
+            std::shared_ptr<ParserKeyword> keyword = (*iter).second;
+            stream << keyword->createDeclaration("   ") << std::endl;
+        }
+        stream << "}" << std::endl << "}" << std::endl;
+        stream << "#endif" << std::endl;
+
+        {
+            bool update = updateFile( stream , headerFile );
+            if (m_verbose) {
+                if (update)
+                    std::cout << "Updated header file written to: " << headerFile << std::endl;
+                else
+                    std::cout << "No changes to header file: " << headerFile << std::endl;
+            }
+            return update;
+        }
+    }
+
+
+    std::string KeywordGenerator::startTest(const std::string& keyword_name) {
+        return std::string("BOOST_AUTO_TEST_CASE(TEST") + keyword_name + std::string("Keyword) {\n");
+    }
+
+
+    std::string KeywordGenerator::endTest() {
+        return "}\n\n";
+    }
+
+
+
+    bool KeywordGenerator::updateTest(const KeywordLoader& loader , const std::string& testFile) const {
+        std::stringstream stream;
+
+        stream << testHeader();
+        for (auto iter = loader.keyword_begin(); iter != loader.keyword_end(); ++iter) {
+            const std::string& keywordName = (*iter).first;
+            std::shared_ptr<ParserKeyword> keyword = (*iter).second;
+            stream << startTest(keywordName);
+            stream << "    std::string jsonFile = \"" << loader.getJsonFile( keywordName) << "\";" << std::endl;
```

Comment:
I think it would be better to use `\n` here rather than `std::endl`. This code is run for each keyword I assume, and actually flushing to file for every line should be unnecessary, and may take significant time (but I have not measured this).